src/callbacks.jl

```diff
@@ -160,8 +160,22 @@ const sensemap = Dict(:(<=) => '<', :(==) => '=', :(>=) => '>')
 ## Lazy constraints
 export @lazyconstraint
 
-macro lazyconstraint(cbdata, x)
-    cbdata = esc(cbdata)
+macro lazyconstraint(args...)
+    cbdata = esc(args[1])
+    x = args[2]
+    extra = vcat(args[3:end]...)
+    # separate out keyword arguments
+    kwargs = filter(ex->isexpr(ex,:kw), extra) # filtering expressions corresponding to kw args specs
+    extra = filter(ex->!isexpr(ex,:kw), extra) # others
+
+    localcut_val = false # by default, the lazy constraint is global
+    for ex in kwargs
+        kwarg = ex.args[1]
+        if kwarg == :localcut
+            localcut_val = esc(ex.args[2])   # excepted if otherwise specified ...
+        end
```

Comment:
Maybe throw an informative error here if users provide a keyword argument that isn't ``localcut``?